src/goto-checker/multi_path_symex_checker.cpp

```diff
@@ -0,0 +1,238 @@
+/*******************************************************************\
+
+Module: Goto Checker using Bounded Model Checking
+
+Author: Daniel Kroening, Peter Schrammel
+
+\*******************************************************************/
+
+/// \file
+/// Goto Checker using Bounded Model Checking
+
+#include "multi_path_symex_checker.h"
+
+#include <chrono>
+
+#include "bmc_util.h"
+#include "counterexample_beautification.h"
+
+multi_path_symex_checkert::multi_path_symex_checkert(
+  const optionst &options,
+  ui_message_handlert &ui_message_handler,
+  abstract_goto_modelt &goto_model)
+  : multi_path_symex_only_checkert(options, ui_message_handler, goto_model),
+    symex_has_run(false)
+{
+  solver_factoryt solvers(
+    options,
+    ns,
+    ui_message_handler,
+    ui_message_handler.get_ui() == ui_message_handlert::uit::XML_UI);
+  solver = solvers.get_solver();
+  solver->prop_conv().set_message_handler(ui_message_handler);
+}
+
+incremental_goto_checkert::resultt multi_path_symex_checkert::
+operator()(propertiest &properties)
+{
+  resultt result;
```

Comment:
Explicitly initialise? At the moment this reads like returning undefined on line 54 (and regardless it's not clear what the default result is)